### PR TITLE
[interpreter] Implement func.bind

### PR DESCRIPTION
**Pull request by rossberg**
This turns out to be very easy in the interpreter...

### CODE REVIEW

File: test/core/func_bind.wast [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+      (func.bind (type $f2) (f64.const 2)
+      (func.bind (type $f3) (f64.const 1) (ref.func $f)))
+    )
+    (table.set $tf (i32.const 22)
+      (func.bind (type $f2)
+      (func.bind (type $f2) (f64.const 2)
+      (func.bind (type $f3)
+      (func.bind (type $f3) (f64.const 1)
+      (func.bind (type $f4)
+      (func.bind (type $f4) (ref.func $f)))))))
```
Comment (binji) [inline test/core/func_bind.wast]: I think this pattern would be a bit clearer with `(ref.func $f)` on its own line. That way, it's easier to see that the list of values to the right represent the bound values.

File: test/core/func_bind.wast [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+    (table.set $tp (i32.const 20) (ref.func $p))
+    (table.set $tp (i32.const 21) (func.bind (type $p2) (ref.func $p)))
+
+    ;; Host closures with arity 1
+    (table.set $tp (i32.const 10)
+      (func.bind (type $p1) (f64.const 1) (ref.func $p))
+    )
+    (table.set $tp (i32.const 11)
+      (func.bind (type $p1) (f64.const 1)
+      (func.bind (type $p2) (ref.func $p)))
```
[the remark this answers is not in the document]
Comment (rossberg) [reply test/core/func_bind.wast]: Just to cover the extreme corner cases. :)

File: test/core/func_bind.wast (hunk 1 above)
Comment (rossberg) [reply test/core/func_bind.wast]: Done, and indented.

File: test/core/func_bind.wast [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+      (func.bind (type $f0) (f64.const 3) (f64.const 4)
+      (func.bind (type $f2) (f64.const 2)
+      (func.bind (type $f3) (f64.const 1) (ref.func $f))))
+    )
+    (table.set $tf (i32.const 05)
+      (func.bind (type $f0) (f64.const 4)
+      (func.bind (type $f1) (f64.const 2) (f64.const 3)
+      (func.bind (type $f3) (f64.const 1) (ref.func $f))))
+    )
+    (table.set $tf (i32.const 05)
```
[the remark this answers is not in the document]
Comment (rossberg) [reply test/core/func_bind.wast]: Good catch, fixed.
